variables.tf

```diff
@@ -0,0 +1,210 @@
+    os_sku: (Required) The os that will be running on the vm.
+    os_offer: (Required) Specifies the offer of the image used to create the virtual machines. Changing this forces a new resource to be created.
+    os_version: (Required) Optionally specify an os version for the chosen sku.
+    os_publisher: (Required) Specifies the Publisher of the Marketplace Image this Virtual Machine should be created from. Changing this forces a new resource to be created.
+    os_disk_caching: Optionally change the caching option of the os disk. Defaults to ReadWrite.
+    os_disk_size_gb: Optionally change the size of the os disk. Defaults to be specified by image.
+    os_disk_storage_type: Optionally change the os_disk_storage_type. Defaults to StandardSSD_LRS.
+    zone: Optionally specify an availibility zone for the vm. Values 1, 2 or 3.
+    availability_set_id: Optionally specify an availibility set for the vm. Not compatible with zone.
+    write_accelerator_enabled: Optionally activate write accelaration for the os disk. Can only
+      be activated on Premium disks and caching deactivated. Defaults to false.
+    proximity_placement_group_id: (Optional) The ID of the Proximity Placement Group which the Virtual Machine should be assigned to.
+    tags: Optionally specify tags in as a map.
+  ```
+  DOC
+}
+
+variable "severity_group" {
+  type        = string
+  default     = ""
+  description = "The severity group of the virtual machine."
+}
+
+variable "update_allowed" {
+  type        = bool
+  default     = true
+  description = "Set the tag `Update allowed`. `True` will set `yes`, `false` to `no`."
+}
+
+variable "data_disks" {
+  type = map(object({
+    lun                        = number
+    disk_size_gb               = number
+    caching                    = optional(string, "ReadWrite")
+    create_option              = optional(string, "Empty")
+    storage_account_type       = optional(string, "StandardSSD_LRS")
+    write_accelerator_enabled  = optional(bool, false)
+    on_demand_bursting_enabled = optional(bool, false)
+  }))
+  validation {
+    condition     = length([for v in var.data_disks : v.lun]) == length(distinct([for v in var.data_disks : v.lun]))
+    error_message = "One or more of the lun parameters in the map are duplicates."
+  }
+  validation {
+    condition     = alltrue([for o in var.data_disks : contains(["Standard_LRS", "StandardSSD_LRS", "Premium_LRS", "StandardSSD_ZRS", "Premium_ZRS"], o.storage_account_type)])
+    error_message = "Possible values are Standard_LRS, StandardSSD_LRS, Premium_LRS, StandardSSD_ZRS and Premium_ZRS for storage_account_type"
+  }
+  validation {
+    condition     = (alltrue([for o in var.data_disks : contains(["Premium_LRS", "Premium_ZRS"], o.storage_account_type)]) && alltrue([for o in var.data_disks : o.write_accelerator_enabled == true]) && alltrue([for o in var.data_disks : o.caching == "None"])) || (alltrue([for o in var.data_disks : o.write_accelerator_enabled == false]))
```

Comment:
That condition seems to check only if all or none of the disks have write accelerator. I think there is an issue when only one disk has write accelerator for example